joy/src/joy_node_win.cpp

```diff
@@ -23,76 +23,269 @@ SOFTWARE.
 */
 
 #include "ros/ros.h"
-#include "geometry_msgs/Twist.h"
-#include "std_msgs/Empty.h"
+#include <diagnostic_updater/diagnostic_updater.h>
+#include <sensor_msgs/Joy.h>
 
-#include <windows.h>
-#include <mmsystem.h>
-#define ERR_NODEVICE -1
 
-int findJoystick()
-{
-  JOYINFO joyinfo; 
-  UINT wNumDevs, wDeviceID; 
-  BOOL bDev1Attached, bDev2Attached; 
- 
-  if((wNumDevs = joyGetNumDevs()) == 0) 
-    return ERR_NODEVICE; 
-
-  bDev1Attached = joyGetPos(JOYSTICKID1,&joyinfo) != JOYERR_UNPLUGGED; 
-  bDev2Attached = wNumDevs == 2 && joyGetPos(JOYSTICKID2,&joyinfo) != JOYERR_UNPLUGGED; 
+// platform
+#if defined(_WIN32)
+    #include <windows.h>
```

Comment:
After a quick glance, I didn't see `Windows` specific code. Do we still need to include `Windows.h` to build?

Reply:
You noticed that :-)
This code is fully cross platform; I want to get it make sure it works great using the Windows use case first before proposing Open Robotics to make it the default implementation.  There is an equivalent ROS2 change needed. Perhaps something to bring up at the TSC?

Reply:
I removed the Windows.h